File: Java/fibonacci.java
// Coding interview question: Print the first n Fibonacci numbers

public class fibonacci {
    public static void printFibonacci(int n) {
        int a = 0;
        int b = 1;

        if (n <= 0) {
            System.out.println("[]");
            return;
        }

        System.out.print(a);

        for (int i = 1; i < n; i++) {
            System.out.print(" " + b);
            int next = a + b;
            a = b;
            b = next;
        }

        System.out.println();
    }

    public static void main(String[] args) {
        printFibonacci(8);
        // Output: 0 1 1 2 3 5 8 13

        printFibonacci(1);
        // Output: 0
    }
}

File: JavaScript/fibonacci.js
// Coding interview question: Print the first n Fibonacci numbers

function fibonacci(n) {
    if (n <= 0) {
        return [];
    }

    const result = [0];

    if (n === 1) {
        return result;
    }

    let a = 0;
    let b = 1;

    while (result.length < n) {
        result.push(b);
        const next = a + b;
        a = b;
        b = next;
    }

    return result;
}

console.log(fibonacci(8));
// Output: [0, 1, 1, 2, 3, 5, 8, 13]

console.log(fibonacci(1));
// Output: [0]

// Explanation:
// Start with 0 and 1, then repeatedly add the previous two numbers.
